assets/js/translator.js

```diff
@@ -730,6 +744,131 @@ function translateDoc() {
     }
 }
 
+window.onpopstate = function () {
+    // TODO: Could do the whole restoreChoices here? But would have to call the right translate() functions anyway
+    if(getURLParam('qP').length > 0) {
+        $('#webpage').val(decodeURIComponent(getURLParam('qP')));
+        translateWebpage();
+    }
+};
+
+function translateLink(href) { // eslint-disable-line no-unused-vars
+    $('#webpage').val(href);
+    if(window.history) {
+        // used by window.onpopstate:
+        window.history.pushState({}, document.title, window.location.href);
+    }
+    translateWebpage();
+}
+
+function cleanPage(html) {
+    // Pages like
+    // eslint-disable-next-line max-len
+    // http://www.lapinkansa.fi/sagat/romssa-sami-searvvi-jodiheaddji-daruiduhttinpolitihkka-vaikkuha-ain-olbmuid-guottuide-samiid-birra-15843633/
+    // insert noise using document.write that 1. makes things
+    // enormously slow, and 2. completely mess up styling so e.g. you
+    // have to scroll through a full screen of whitespace before
+    // reaching content. This might mess things up some places – needs
+    // testing – but on the other hand most uses of document.write are
+    // evil.
+    return html.replace(/document[.]write[(]/g,
+        'console.log("document.write "+');
+}
+
+function translateWebpage() {
+    persistChoices('translator', true);
+    if(!$('div#translateWebpage').is(':visible')) {
+        showTranslateWebpageInterface($('#originalText').val().trim());
+    }
+
+    if(pairs[curSrcLang] && pairs[curSrcLang].indexOf(curDstLang) !== -1) {
+        sendEvent('translator', 'translateWebpage', curSrcLang + '-' + curDstLang);
+        $('iframe#translatedWebpage').animate({'opacity': 0.75}, 'fast');
+        $.ajax({
```

Comment:
Um, in the current `callApy` method, it will call `$.jsonp()` for the webpage translate request. In order for it to get called by `ajax`, should I add a condition something like the following?

```

if(requestUrl.length > APY_REQUEST_URL_THRESHOLD_LENGTH || endpoint === 'translatePage') {
    requestOptions.type = 'POST';
    return $.ajax(requestOptions);
}
```

I plan to make an issue as you had mentioned as to why it currently doesn't return the error object with simply `$.jsonp()`.